tinygrad/tensor.py

```diff
@@ -150,16 +150,20 @@ def where(self:Tensor, input_:Union[Tensor, float], other:Union[Tensor, float]):
   # ***** (numpy) rng helper functions *****
   # TODO: move randomness generation out of numpy
 
-  _rng: ClassVar[np.random.Generator] = np.random.default_rng()
+  _rng: ClassVar[Union[np.random.Generator, np.random.RandomState]] = np.random.default_rng()
```

Comment:
Why do we need two different ones? Anything wrong with just changing it?

Reply:
The methods in those classes may have different signatures. See for example [RandomState.standard_normal](https://numpy.org/doc/stable/reference/random/generated/numpy.random.RandomState.standard_normal.html#numpy.random.RandomState.standard_normal) vs [Generator.standard_normal](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.standard_normal.html#numpy.random.Generator.standard_normal)

If you just assign a `RandomState` to `Tensor._rng` someone could call methods on it thinking they can use `Generator`'s signatures when they actually can't.

Reply:
If instead by "just changing it" you meant keeping only the legacy one, I think it'd be quite weird to force the worse implementation because of ONNX tests suite requirements.

Reply:
Why is it worse?

Reply:
See here: https://numpy.org/doc/stable/reference/random/legacy.html#numpy.random.RandomState

> Container for the _slow_ Mersenne Twister pseudo-random number generator. Consider using a different BitGenerator with the Generator container instead.

Also here: https://numpy.org/doc/stable/reference/random/legacy.html#legacy-random-generation

> The [RandomState](https://numpy.org/doc/stable/reference/random/legacy.html#numpy.random.RandomState) provides access to legacy generators. [...] This class should only be used if it is essential to have randoms that are identical to what would have been produced by previous versions of NumPy.